### PR TITLE
Feature: Improved the display of keyboard shortcuts in the Command Palette

### LINKED ISSUE #13200: Feature: Add KeyboardShortcut custom control
# Feature: Add KeyboardShortcut custom control

## Motivation and Context

Currently the Files displays keyboard shortcuts in TextBlocks and gives users no visual effect. Following the new keyboard shortcut custom control that Windows introduced in Windows 11, we introduce a new control. This can be used in the hotkey customization settings page that may be implemented in the future, in the command palette suggestion items, and even in the MenuFlyout items.

**I got inspiration from this UI**
image

**Ugly look in the command palette**
image

### Development & co-authors

- #13201
- @0x5bfa 

# Specs

### Definition
```cs
public class KeyboardShortcut : Control
```

### Description
Handles KeyboardShortcut control comprehensively.

### Status
- [x] Drafted
- [ ] Implemented

### Properties
Property name|Type|Description
---|---|---
ShortcutText|string|Gets or sets the content that represents the keyboard shortcut to be displayed in the control

### Pseudo code template (not actual usage)
```xaml
<></>
```

### Remarks
_None_

# Concepts

Made by @mdtauk

**Default style**
image

**Accent style**
image

**Reveal style**
image

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

As the title.

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13200
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open Command Palette with no words

## Screenshots

image

### CODE REVIEW

File: src/Files.App/UserControls/AddressToolbar.xaml [hunk 1]
```diff
@@ -313,34 +316,46 @@
...
+									TextTrimming="CharacterEllipsis"
+									TextWrapping="NoWrap">
 									<Run FontWeight="Normal" Text="{x:Bind PrimaryDisplayPreMatched, Mode=OneWay}" /><Run FontWeight="Bold" Text="{x:Bind PrimaryDisplayMatched, Mode=OneWay}" /><Run FontWeight="Normal" Text="{x:Bind PrimaryDisplayPostMatched, Mode=OneWay}" />
+									<!--<ToolTipService.ToolTip>
+										<TextBlock>
+											<Run Text="{x:Bind PrimaryDisplay, Mode=OneWay}" />
+											<Run Text="&#10;" />
+											<Run FontStyle="Italic" Text="Ctrl+A" />
+										</TextBlock>
+									</ToolTipService.ToolTip>-->
```
Comment (yaira2) [inline src/Files.App/UserControls/AddressToolbar.xaml:340]: What's the tooltip for?

Comment (0x5bfa) [reply src/Files.App/UserControls/AddressToolbar.xaml]: I'm not sure, deleted!

**Comment by 0x5bfa** [comment]
@hishitetsu Is secondary display and supplymentary display used no longer in this PR? If so, I'd like to remove them in this PR.

**Comment by hishitetsu** [comment]
They were originally intended to be displayed on two lines, one in the native language and one in English, as in VSCode. If we don't plan to do so, you may remove them.
image

**Comment by 0x5bfa** [comment]
Thanks. I will remove for now.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
@0x5bfa can you try to center the text vertically?
image

**Comment by 0x5bfa** [comment]
Two hours please I have something else to do irl.

File: src/Files.App/UserControls/KeyboardShortcut/KeyboardShortcut.xaml [hunk 2]
```diff
@@ -4,46 +4,60 @@
...
 	xmlns:local="using:Files.App.UserControls.KeyboardShortcut">
 
-	<local:KeyboardShortcutItemStyleSelector
-		x:Key="KeyboardShortcutItemStyleSelector"
-		AccentItemStyle="{StaticResource AccentKeyboardShortcutItemStyle}"
-		DefaultItemStyle="{StaticResource DefaultKeyboardShortcutItemStyle}"
-		RevealItemStyle="{StaticResource RevealKeyboardShortcutItemStyle}" />
+	<ResourceDictionary.ThemeDictionaries>
+		<ResourceDictionary x:Key="Light">
+			<StaticResource x:Key="OutlinedStyleBackgroundBrush" ResourceKey="ControlFillColorTransparentBrush" />
```
Comment (yaira2) [inline src/Files.App/UserControls/KeyboardShortcut/KeyboardShortcut.xaml:9]: Shouldn't this be defined as a SolidColorBrush instead of StaticResource?

Comment (0x5bfa) [reply src/Files.App/UserControls/KeyboardShortcut/KeyboardShortcut.xaml:9]: I'm not sure what the difference is between SolidColorBrush and StaticResource.
Should it be? I saw in this case using StaticResource in Button_themeresources.xaml in MUXC 1.4

**Pull request merged**